src/_pyinstaller_hooks_contrib/hooks/stdhooks/hook-pyarrow.py

```diff
@@ -12,13 +12,8 @@
 
 # Hook for https://pypi.org/project/pyarrow/
 
-from PyInstaller.utils.hooks import collect_data_files, collect_dynamic_libs
-
-hiddenimports = [
-    "pyarrow._parquet",
-    "pyarrow.lib",
-    "pyarrow.compat",
-]
+from PyInstaller.utils.hooks import collect_submodules, collect_data_files, collect_dynamic_libs
 
+hiddenimports = collect_submodules('pyarrow', filter=lambda x: "tests" not in x.split("."))
```

Comment:
Splitting the name string is redundant here, isn't it?

Reply:
You are right, I removed the split